HotelReservation/HotelReservation/ReservationSteps/StepsExecutor.cs

```diff
@@ -7,19 +7,19 @@ namespace HotelReservation.ReservationSteps
 {
     public class StepsExecutor
     {
-        private readonly Dictionary<ReservationSteps, IReservationStep> _reservationStepsInstances;
+        private readonly Dictionary<ReservationStepType, IReservationStep> _reservationStepsInstances;
 
         public StepsExecutor()
         {
-            _reservationStepsInstances = new Dictionary<ReservationSteps, IReservationStep>
+            _reservationStepsInstances = new Dictionary<ReservationStepType, IReservationStep>
                 {
-                    {ReservationSteps.ReservationProcess, new ReservationStartProcess() },
-                    {ReservationSteps.SendingMailProcess, new SendingMailProcess() },
-                    {ReservationSteps.PaymentProcess, new PaymentProcess() }
+                    {ReservationStepType.ReservationProcess, new ReservationStartProcess() },
```

Comment:
Kod się już trochę uprościł, więc wróćmy do tematu factory, którego kiedyś było dużo w projekcie

Executor wykonuje kroki zawsze tak samo (i dobrze). Nie powinien się on więc zmieniać w sytuacji, gdy dodajemy nowy krok. Warto by było wydelegować tworzenie intancji dla danego typu do osobnej klasy.
Drugą sprawą, która póki co ma małe znaczenie, jest tworzenie wszystkich instancji zanim ich potrzebujesz. Warto tworzyć takie obiekty tuż przed Execute() zmiast trzymać na zawsze w pamięci.

Reply:
Fixed - instancje tworzone w klasie StepFactory i zwracane przez metodę CreateInstance()

Reply:
> Drugą sprawą, która póki co ma małe znaczenie, jest tworzenie wszystkich instancji zanim ich potrzebujesz. Warto tworzyć takie obiekty tuż przed Execute() zmiast trzymać na zawsze w pamięci.

Zrobiłeś właśnie tak, że inicjalizujesz wszystkie stepy. Daj znać, czy to, co napisałem jest niezrozumiałe, czy celowo ominąłeś sugestię

Reply:
> > Drugą sprawą, która póki co ma małe znaczenie, jest tworzenie wszystkich instancji zanim ich potrzebujesz. Warto tworzyć takie obiekty tuż przed Execute() zmiast trzymać na zawsze w pamięci.
> 
> Zrobiłeś właśnie tak, że inicjalizujesz wszystkie stepy. Daj znać, czy to, co napisałem jest niezrozumiałe, czy celowo ominąłeś sugestię

Fixed - pominąłem, bo zrozumiałem, że skoro nie jest na tę chwilę ważne, to nie powinienem się tym zajmować :) przy okazji pytanie - co myślisz o używaniu "Lazy"?